pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -271,6 +271,42 @@ impl<T: Config> Pallet<T> {
 		fee
 	}
 
+	/// Compute the length portion of a fee by invoking the configured `LengthToFee` impl.
+	/// # Arguments
+	/// * `length` - The length of the transaction.
+	/// * `weight` - The weight of the transaction.
+	///
+	/// # Returns
+	/// `FeeDetails` - The fee details for the transaction.
+	pub fn compute_capacity_fee_details<
+		Extrinsic: sp_runtime::traits::Extrinsic + GetDispatchInfo,
+	>(
+		unchecked_extrinsic: Extrinsic,
+		len: u32,
+	) -> FeeDetails<BalanceOf<T>>
+	where
+		<T as pallet::Config>::RuntimeCall: Dispatchable<Info = DispatchInfo>,
+	{
+		let dispatch_info = <Extrinsic as GetDispatchInfo>::get_dispatch_info(&unchecked_extrinsic);
+		let extrinsic_weight = dispatch_info.weight;
```

Comment:
Perhaps, I am wrong but `get_dispatch_info` is not using the stable weights.
Here you need to grab the stable weight.

Reply:
Using stable weights